win-spout.cpp

```diff
@@ -54,40 +54,46 @@ struct win_spout {
 
 	SPOUTHANDLE spoutptr;
 
-	DWORD lastCheckTick;
+	ULONGLONG lastCheckTick;
 
 	int width;
 	int height;
 
 	bool initialized;
 	bool active;
+
+	int tickspeedlimit;
+
+	int spout_status;
+	int render_status;
+	int tick_status;
+	bool need_release;
 };
-// gets the first spout sender available -- assumes that sender name
-// is maximum size 256 (default for spout SDK)
-static bool get_first_spount_sender(SPOUTHANDLE spoutptr, char *sender_name)
+
+/**
+ * Updates sender texture details on the context
+ * @return bool success
+ */
+static bool win_spout_get_sender_info(win_spout * context)
 {
-	int totalSenders;
-	totalSenders = spoutptr->GetSenderCount();
-	if (totalSenders == 0) {
+	unsigned int width, height;
+	// get info about this active sender:
+	if (!context->spoutptr->GetSenderInfo(context->senderName, width,
+					      height, context->dxHandle,
+					      context->dxFormat)) {
 		return false;
 	}
-	if (!spoutptr->GetSenderName(0, sender_name))
-		return false;
-
-	blog(LOG_INFO, "Sender name %s, total senders %d", sender_name,
-	     totalSenders);
-
-	if (!spoutptr->SetActiveSender(sender_name))
-		return false;
 
+	context->width = width;
+	context->height = height;
 	return true;
 }
 
 /**
- * Updates sender texture details on the context
+ * Updates and Log sender texture details on the context
  * @return bool success
  */
-static bool win_spout_get_sender_info(win_spout * context)
+static bool win_spout_log_sender_info(win_spout *context)
```

Comment:
Main purpose of this function is to set the `context->width` ,`context->height` and so it is kind of `getting`, but it's also getting the `dxFormat` and the `dxHandle`. 

I probably have too much logging here ;)

```suggestion
static bool win_spout_store_sender_info(win_spout *context)
```

.... from the Spout SDK Manual:
![image](https://user-images.githubusercontent.com/1377367/75723706-b5bb6200-5cd4-11ea-8802-cabb323553f3.png)

Reply:
win_spout_get_sender_info and win_spout_log_sender_info are same function . the only different is whether they will output log or not.

Reply:
Yes. I can see you've just renamed the function. My point was that the function does not just log info, it also is storing information about the sender to the context.

I am open to more suggestions about ways to store that sender information.

Is that what you meant?